src/Hateoas/Serializer/Metadata/RelationPropertyMetadata.php

```diff
@@ -34,5 +35,11 @@ public function __construct(Exclusion $exclusion = null, Relation $relation = nu
         $this->sinceVersion = $exclusion->getSinceVersion();
         $this->untilVersion = $exclusion->getUntilVersion();
         $this->maxDepth = $exclusion->getMaxDepth();
+
+        if ($exclusion->getExcludeIf() !== null && preg_match(self::EXPRESSION_REGEX, $exclusion->getExcludeIf(), $matches)) {
```

Comment:
This logic is now duplicated 3 times. Why did you remove the ExpressionEvaluator class ?

Reply:
indeed, this will be solved in a separate PR (see https://github.com/willdurand/Hateoas/issues/270)